components/binary_sensor/xiaomi.py

```diff
@@ -6,67 +6,46 @@
 import logging
 import asyncio
 
-from homeassistant.helpers.entity import Entity
-from homeassistant.components.binary_sensor import (BinarySensorDevice)
-from homeassistant.const import ATTR_BATTERY_LEVEL
+from homeassistant.components.binary_sensor import BinarySensorDevice
+try:
+    from homeassistant.components.xiaomi import XiaomiDevice
+except ImportError:
+    from custom_components.xiaomi import XiaomiDevice
 
 _LOGGER = logging.getLogger(__name__)
 
+
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Perform the setup for Xiaomi devices."""
-
+    devices = []

```

Comment:
HA is optimized for receiving all the sensor at once. This will improve the startup time of HA